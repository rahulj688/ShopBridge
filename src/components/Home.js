import React, { Component } from 'react';
import Inventory from './Inventory';

export default class Home extends React.PureComponent {
  constructor() {
    super();
    this.state = {
      imgFile: ''
    }

  }

  getCurrentDetails = async () => {
    await fetch('https://5f44abf43fb92f0016753a78.mockapi.io/products',
      {
        method: 'GET'
      })
      .then(async (res) => {
        let response = await res.json();
        console.log("res: ", response)
        this.setState({ 
          inventPro: response
        })
        return response;
      })
      .catch((error) => {
      })
  }

  fileSelected = (file) => {
    this.setState({ productDetails: '' })
    var reader = new FileReader(); 
    var fileContent;
    reader.readAsDataURL(file[0]);
    var me = this; 
    reader.onload = () => { 
      fileContent = reader.result; 
      console.log(fileContent);
      this.setState({
        image: fileContent
      })
    }
  }

  addPrdouctToInventory = async () => {
    // let inventoryData = [];
    // let length;
    // fetch('https://5f44abf43fb92f0016753a78.mockapi.io/products',
    //   {
    //     method: 'GET'
    //   })
    //   .then(async (res) => {
    //     let response = await res.json();
    //     console.log("res: ", response)
    //     inventoryData = response;
    //     let length  = inventoryData.length;
    //   })
    //   .catch((error) => {
    //   })
    await this.getCurrentDetails();
      let lastId = this.state.inventPro.length > 0 ? this.state.inventPro[this.state.inventPro.length - 1].id : 0;
      console.log("image: ", this.state.image);
      let product = 
        {
          "id": (parseInt(lastId) + 1).toString(),
        "name": this.state.name,
        "image": this.state.image ? this.state.image : require('../assets/logo.jpg'),
        "description": this.state.description,
        "price": parseInt(this.state.price)
        }
        
       fetch('https://5f44abf43fb92f0016753a78.mockapi.io/products',
        {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(product)
        })
        .then((data) => {
          let d = data;
          console.log('submit: ', d)
        })
        .catch((error) => {
          console.log('err: ', error)
        })
        this.setState({
          name: '',
          description: '',
          price: '',
          fileInput: '',
          productDetails: product,
          updateTable: true
        })
    
  }

  updateValue = () => {
    this.setState({updateTable: false})
  }

  render() {
    return (
      <div style={{ backgroundColor: 'lightgray', height: 200, marginLeft: 20, marginRight: 20, borderRadius: 10 }}>
        <div style={{ marginTop: 10 }}>
          <h3 style={{ marginTop: 10, marginLeft: 20 }}>Add a New Product to Inventory</h3>
          <div style={{ display: 'flex', flexDirection: 'row', marginTop: 30 }}>
            <form>
              <label htmlFor="name" style={{ marginLeft: 30 }}>Name:</label>
              <input type="text" value={this.state.name} placeholder="Name of Product" required style={{ height: 25, marginLeft: 10, marginTop: 3, borderRadius: 3 }} onChange={(e) => this.setState({ name: e.target.value, productDetails: '' })} />

              <label htmlFor="description" style={{ marginLeft: 20 }}>Description:</label>
              <textarea value={this.state.description} placeholder="Enter Description of Product separated by comma(,)" rows="3" required cols="30" style={{ marginLeft: 10, borderRadius: 5 }} onChange={(e) => this.setState({ description: e.target.value, productDetails: '' })} ></textarea>

              <label htmlFor="price" style={{ marginLeft: 20 }}>Price:</label>
              <input type="text" value={this.state.price} pattern="[0-9]+" placeholder="Price of Product" required style={{ height: 25, marginLeft: 10, marginTop: 3, borderRadius: 3 }} onChange={(e) => this.setState({ price: e.target.value, productDetails: '' })} />

              <label htmlFor="myfile" style={{ marginLeft: 20 }} >Select a file:</label>
              <input type="file" accept="image/*" id="myfile" value={this.state.fileInput} style={{ marginLeft: 10, width: 290 }} name="myfile" onChange={(e) => this.fileSelected(e.target.files)} />
              <input type="button" onClick={() => this.addPrdouctToInventory()} value="Submit" disabled={(this.state.name && this.state.price && this.state.description) ? false : true} />
            </form>
          </div>
        </div>
          <Inventory productDetail={this.state.productDetails} updateTable={this.state.updateTable} update={() => this.updateValue()} />
      </div>
    )
  }
}
