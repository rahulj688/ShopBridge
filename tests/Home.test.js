import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Home from '../src/components/Home';
 
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Home />, div);
    ReactDOM.unmountComponentAtNode(div);
});
it('renders correctly', () => {
  const h = renderer.create(<Home />).toJSON();
  h.addPrdouctToInventory();
  expect(h).toMatchSnapshot();
});
