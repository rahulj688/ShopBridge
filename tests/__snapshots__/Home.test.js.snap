exports[`test renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightgray",
      "borderRadius": 10,
      "height": 200,
      "marginLeft": 50,
      "marginRight": 50,
    }
  }>
  <div
    style={
      Object {
        "marginTop": 10,
      }
    }>
    <h3
      style={
        Object {
          "marginLeft": 20,
          "marginTop": 10,
        }
      }>
      Add a New Product to Inventory
    </h3>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 30,
        }
      }>
      <label
        htmlFor="name"
        style={
          Object {
            "marginLeft": 30,
          }
        }>
        Name:
      </label>
      <input
        onChange={[Function]}
        placeholder="Name of Product"
        required={true}
        style={
          Object {
            "borderRadius": 3,
            "height": 25,
            "marginLeft": 10,
            "marginTop": 3,
          }
        }
        type="text" />
      <label
        htmlFor="description"
        style={
          Object {
            "marginLeft": 20,
          }
        }>
        Description:
      </label>
      <textarea
        cols="30"
        onChange={[Function]}
        placeholder="Enter Description of Product separated by comma(,)"
        required={true}
        rows="3"
        style={
          Object {
            "borderRadius": 5,
            "marginLeft": 10,
          }
        } />
      <label
        htmlFor="price"
        style={
          Object {
            "marginLeft": 20,
          }
        }>
        Price:
      </label>
      <input
        onChange={[Function]}
        placeholder="Price of Product"
        required={true}
        style={
          Object {
            "borderRadius": 3,
            "height": 25,
            "marginLeft": 10,
            "marginTop": 3,
          }
        }
        type="text" />
      <label
        htmlFor="myfile"
        style={
          Object {
            "marginLeft": 20,
            "width": 100,
          }
        }>
        Select a file:
      </label>
      <input
        id="myfile"
        name="myfile"
        onChange={[Function]}
        required={true}
        style={
          Object {
            "marginLeft": 10,
          }
        }
        type="file" />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "marginLeft": 650,
          "marginTop": 10,
        }
      }
      type="button">
      Submit
    </button>
  </div>
  
  <div
    style={
      Object {
        "marginTop": 80,
        "textAlign": "center",
      }
    }>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }>
      <h4
        style={
          Object {
            "textAlign": "left",
          }
        }>
        Inventory Products
      </h4>
    </div>
    <table
      className="table table-striped table-bordered table-hover"
      style={
        Object {
          "marginTop": 20,
        }
      }>
      <thead>
        <tr>
          <th>
            Product ID
          </th>
          <th>
            Product Image
          </th>
          <th>
            Product Name
          </th>
          <th>
            Product Price
          </th>
          <th>
            Delete
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              onClick={[Function]}>
              1
            </div>
          </td>
          <td>
            <img
              alt="image"
              height="100"
              src="image1.jpg"
              weight="100" />
          </td>
          <td>
            Highlander White Slim Fit Casual Shirt
          </td>
          <td>
            800
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button">
              <svg
                className={undefined}
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg">
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
                </g>
              </svg>
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              onClick={[Function]}>
              2
            </div>
          </td>
          <td>
            <img
              alt="image"
              height="100"
              src="image2.jpg"
              weight="100" />
          </td>
          <td>
            Men Red & Black Regular Fit Checked Casual Shirt
          </td>
          <td>
            1000
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button">
              <svg
                className={undefined}
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg">
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
                </g>
              </svg>
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              onClick={[Function]}>
              3
            </div>
          </td>
          <td>
            <img
              alt="image"
              height="100"
              src="image3.jpeg"
              weight="100" />
          </td>
          <td>
            Men Black Slim Fit Solid Casual Shirt
          </td>
          <td>
            900
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button">
              <svg
                className={undefined}
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg">
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
                </g>
              </svg>
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              onClick={[Function]}>
              4
            </div>
          </td>
          <td>
            <img
              alt="image"
              height="100"
              src="image4.jpg"
              weight="100" />
          </td>
          <td>
            Men Yellow Printed Round Neck T-Shirt
          </td>
          <td>
            750
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button">
              <svg
                className={undefined}
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg">
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
                </g>
              </svg>
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <div
              onClick={[Function]}>
              5
            </div>
          </td>
          <td>
            <img
              alt="image"
              height="100"
              src="image5.jpg"
              weight="100" />
          </td>
          <td>
            Men Navy Blue Tapered Fit Mid-Rise Clean Look Stretchable Jeans
          </td>
          <td>
            1300
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button">
              <svg
                className={undefined}
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg">
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none" />
                  <path
                    d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
                </g>
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
